autogpt_platform/backend/backend/copilot/executor/processor.py

```
@@ -518,48 +519,112 @@ async def _execute_async(
+                    credential_lease = await asyncio.wait_for(
+                        IntegrationCredentialsManager().acquire_lease(
+                            codex_user_id,
+                            entry.llm_credential_id,
+                        ),
+                        timeout=_CODEX_CREDENTIAL_ACQUIRE_TIMEOUT_SECONDS,
+                    )
+                except asyncio.TimeoutError:
+                    raise RuntimeError("codex_credential_busy") from None
+                except asyncio.CancelledError:
+                    raise
+                except Exception:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)